stacks/ts/src/ntt.ts

```diff
@@ -0,0 +1,861 @@
+import {
+  Chain,
+  encoding,
+  Network,
+  toChainId,
+} from "@wormhole-foundation/sdk-base";
+import {
+  AccountAddress,
+  ChainAddress,
+  ChainsConfig,
+  Contracts,
+  serialize,
+  toNative,
+  UniversalAddress,
+} from "@wormhole-foundation/sdk-definitions";
+import { Ntt, NttTransceiver } from "@wormhole-foundation/sdk-definitions-ntt";
+import {
+  StacksChains,
+  StacksPlatform,
+  StacksPlatformType,
+  StacksUnsignedTransaction,
+  StacksZeroAddress,
+} from "@wormhole-foundation/sdk-stacks";
+import { StacksNetwork } from "@stacks/network";
+import {
+  BufferCV,
+  Cl,
+  cvToValue,
+  fetchCallReadOnlyFunction,
+  PostConditionMode,
+} from "@stacks/transactions";
+import {
+  ADDR32_CONTRACT,
+  DEFAULT_NTT_VERSION,
+  NTT_MANAGER_CONTRACT_NAME,
+  NTT_MANAGER_STATE_CONTRACT_NAME,
+  NTT_TOKEN_OWNER_CONTRACT_NAME,
+  WORMHOLE_PROTOCOL_ID,
+  WORMHOLE_TRANSCEIVER_STATE_CONTRACT_NAME,
+} from "./constants.js";
+
+export class StacksNttWormholeTransceiver<
+  N extends Network,
+  C extends StacksChains,
+> implements NttTransceiver<N, C, Ntt.Attestation>
+{
+  private readonly contractName: string;
+  private readonly deployer: string;
+
+  constructor(
+    readonly manager: StacksNtt<N, C>,
+    readonly tokenAddress: string,
+    readonly address: string
+  ) {
+    const addressSplit = address.split(".");
+    this.contractName = addressSplit[1]!;
+    this.deployer = addressSplit[0]!;
+  }
+
+  async getTransceiverType(): Promise<string> {
+    return "wormhole";
+  }
+
+  async getAddress(): Promise<ChainAddress<C>> {
+    return {
+      chain: this.manager.chain,
+      address: toNative(this.manager.chain, this.address),
+    };
+  }
+
+  async *setPeer(
```

Comment:
Some of the functions in here were written by @fergarrui who will be adding CLI support in the future.